src/main/java/pl/onsight/wypozyczalnia/model/entity/ProductOrderEntity.java

```diff
@@ -94,4 +138,48 @@ public void setProducts(List<ProductEntity> products) {
     public ProductOrderEntity(List<ProductEntity> products) {
         this.products = products;
     }
+
+    public int getNumberOfDays() {
+        return (orderStart != null && orderEnd != null) ? Days.daysBetween(new DateTime(orderStart), new DateTime(orderEnd)).getDays() + 1 : 0;
+    }
+    private void createOldPricesHashMap(Cart cart) {
+        for (ProductEntity product : cart.getProducts()) {
+            oldPrices.put(product.getId(), product.getPrice());
+        }
+    }
+
+    public int countNumberOfProduct(ProductEntity product){
+        return Collections.frequency(products, product);
+
+    }
+
+    public void buildOrder(UserEntity user, Cart cart) {
+
+
+        this.setUser(user);
+        this.setProducts(cart.getProducts());
+        this.setOrderStart(DateFilter.changeStringToDate(cart.getDate())[0]);
+        this.setOrderEnd(DateFilter.changeStringToDate(cart.getDate())[1]);
+        this.setCombinedPrice(cart.getPriceWithDiscount(user));
+        this.setDeposit(cart.getCombinedDeposit());
+        this.setCombinedDiscount(user.getRole().getDiscount());
+        this.createOldPricesHashMap(cart);
+    }
+
+    @Override
+    public String toString() {
+        return "ProductOrderEntity{" +
+                "id=" + id +
+                ", user=" + user +
+                ", products=" + products +
+                ", oldPrices=" + oldPrices +
+                ", orderStart=" + orderStart +
+                ", orderEnd=" + orderEnd +
+                ", combinedPrice=" + combinedPrice +
+                ", deposit=" + deposit +
+                ", paid=" + paid +
+                ", status='" + status + '\'' +
+                ", combinedDiscount=" + combinedDiscount +
+                '}';
+    }
```

Comment:
ale się rozrosła ta klasa :v

Reply:
Zastanawialem sie jeszcze czy nie zapisac jeszcze poszczególnych wartości kaucji ale to jednak niepotrzebne :) juz sama kaucja łączna wystaczy :P